frontend/src/libs/components/note/note.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  changeCursorPosition,
+  debounce,
+  getCursorPosition,
+  getValidClassNames,
+  sanitizeInput,
+} from '#libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useFormController,
+  useParams,
+  useRef,
+} from '#libs/hooks/hooks.js';
+import { type UserAuthResponseDto } from '#packages/users/users.js';
+import { actions as journalActions } from '#slices/journal/journal.js';
+
+import { NOTE_TIMEOUT } from './libs/constants.js';
+import { type NoteContent } from './libs/types.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  className: string;
+};
+
+const Note: React.FC<Properties> = ({ className }) => {
+  const { userId, selectedJournalEntry } = useAppSelector(
+    ({ auth, journal }) => {
+      return {
+        userId: (auth.authenticatedUser as UserAuthResponseDto).id,
+        selectedJournalEntry: journal.selectedJournalEntry,
+      };
+    },
+  );
+
+  const { isDirty, control } = useAppForm({
+    defaultValues: {
+      title: selectedJournalEntry?.title,
+      text: selectedJournalEntry?.text,
+    },
+    mode: 'onChange',
+  });
+
+  const { field: titleField } = useFormController({ name: 'title', control });
+  const { field: textField } = useFormController({ name: 'text', control });
+  const { value: titleValue, onChange: onTitleChange } = titleField;
+  const { value: textValue, onChange: onTextChange } = textField;
+
+  const dispatch = useAppDispatch();
+  const { id } = useParams();
+
+  const titleReference = useRef<HTMLDivElement | null>(null);
+  const textReference = useRef<HTMLDivElement | null>(null);
+  const cursorPosition = useRef<number | null>(null);
+
+  const handleSaveNote = useCallback(
+    (data: NoteContent) => {
+      if (data.title && data.text && id) {
```

Comment:
we need to save it even there is no title or text, also let's pass id in params, not to check it's presence twice
```suggestion
  const handleSaveNote = useCallback(
    (id: number, data: NoteContent) => {
```

Reply:
But if we send empty note data (without title and text) we might have many empty cards in the sidebar, I think it will not be OK.